### PR TITLE
Yosemite: add Fees to Orders

### PR DESCRIPTION
**Pull request by ctarda**
This is the third step to add support for Fees to the Order Details page. (#3084). 
* Storage: #3461 
* Networking: #3452 

This PR implements support for Fees in Yosemite. Support for Fees in the UI will be added in a subsequent PR.

## Changes
* Implement extension for compliance with readonlyConvertible. 

## How to test
* 👀  on the code
* ✅ on CI

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Model/Storage/OrderFeeLine+ReadOnlyConvertible.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+extension Storage.OrderFeeLine: ReadOnlyConvertible {
+
+    /// Updates the Storage.OrderFeeLine with the ReadOnly.
+    ///
+    public func update(with feeLine: Yosemite.OrderFeeLine) {
+        feeID = feeLine.feeID
+        name = feeLine.name
+        taxClass = feeLine.taxClass
+        taxStatusKey = feeLine.taxStatus.rawValue
+        total = feeLine.total
```
Comment (jaclync) [inline Yosemite/Yosemite/Model/Storage/OrderFeeLine+ReadOnlyConvertible.swift:16]: do we also want to update `totalTax` here?

File: Yosemite/Yosemite/Model/Storage/OrderFeeLine+ReadOnlyConvertible.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        let feeAttributes = attributes?.map { $0.toReadOnly() } ?? [Yosemite.OrderItemAttribute]()
+
+        return OrderFeeLine(feeID: feeID,
+                            name: name ?? "",
+                            taxClass: taxClass ?? "",
+                            taxStatus: OrderFeeTaxStatus(rawValue: taxStatusKey),
+                                total: total ?? "",
+                                totalTax: totalTax ?? "",
+                                taxes: feeTaxes,
+                                attributes: feeAttributes)
```
Comment (jaclync) [inline Yosemite/Yosemite/Model/Storage/OrderFeeLine+ReadOnlyConvertible.swift]: nit: would be nice to fix the indentation, I thought these are nested properties 😄

**Review by jaclync — APPROVED** [review]
Do we also need to upsert the fees when upserting an order in `OrdersUpsertUseCase`, or this is part of the next PR? If so, feel free to merge once the question about updating `OrderFeeLine.totalTax` is addressed!

**Comment by ctarda** [comment]
Thanks for the feedback @jaclync I just pushed an update incorporating your feedback, will merge after CI is done

**Pull request merged**